src/common/options.cc

```diff
@@ -605,7 +605,7 @@ std::vector<Option> get_global_options() {
 
     Option("err_to_stderr", Option::TYPE_BOOL, Option::LEVEL_BASIC)
     .set_default(false)
-    .set_daemon_default(true)
+    .set_daemon_default(false)
```

Comment:
If the specific reason is to avoid putting this in the systemd journal, perhaps we could pass --no-err-to-stderr in the systemd unit files?  The only concern I have is that this helps with teuthology runs because we see these errors in teuthology.log.  We could set err_to_stderr in teuthology, though.  So I don't have a strong opinion either way.

There are lots of derr lines in the code. It would be nice to retain the option to send them to the journal (as kefu says).  Can you pick out the specific ones that shouldn't be there and maybe we can change those to dout(0)?

Reply:
we actually see lots of logs as below in the syslog file:
```
Sep 4 10:07:50 cmct-ceph3 ceph-osd[617214]: 2018-09-04 04:07:50.376961 7f1e5c03e700 -1 osd.0 445 heartbeat_check: no reply from 111.111.9.2:6808 osd.3 since back 2018-09-04 04:07:49.246845 front 2018-09-04 04:07:49.246845 (oldest deadline 2018-09-04 04:07:50.213366)
Sep 4 10:07:50 cmct-ceph3 ceph-osd[617193]: 2018-09-04 04:07:50.408334 7f5ed419b700 -1 osd.2 445 heartbeat_check: no reply from 111.111.9.2:6808 osd.3 since back 2018-09-04 04:07:49.189779 front 2018-09-04 04:07:49.189779 (oldest deadline 2018-09-04 04:07:50.174806)
Sep 4 10:07:50 cmct-ceph3 ceph-osd[617214]: 2018-09-04 04:07:50.477094 7f1e5c03e700 -1 osd.0 445 heartbeat_check: no reply from 111.111.9.2:6808 osd.3 since back 2018-09-04 04:07:49.246845 front 2018-09-04 04:07:49.246845 (oldest deadline 2018-09-04 04:07:50.213366)
```
but it's a real error log, so I didn't pick it out to change.  on the contrary, I picked  two trivial logs that shouldn't be there.

another commit `global: fix err_to_stderr can't work when it was configured to false`  seems to be a real fix.

so can you recheck it? @liewegas